src/foam/net/ip/IPAddressInfo.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ */
+
+ foam.CLASS({
+  package: 'foam.net.ip',
+  name: 'IPAddressInfo',
+  documentation: `Represents information fetched by an IP information provider.`,
+  ids: ['provider'],
```

Comment:
Is this really the primary key? This means you can only have one entry per provider.